cypress/integration/instance/InstanceCreation.ts

```diff
@@ -59,6 +59,24 @@ describe("the 'Create a SE instance' Modal", () => {
           "have.text",
           "0 of 3 steps completed"
         );
+
+        // Prepare cypress for 'non deterministic' updates in 'Status' popover
+        // The 'deterministic' updates would be: (0 of 3, 1 of 3, 2 of 3 and Done)
+        // However in reality it may happen: (0 of 3, 1 of 3 and Done) or (0 of 3, 2 of 3 and Done) or ...
+        // see https://github.com/cypress-io/cypress-example-recipes/blob/master/examples/fundamentals__errors/cypress/e2e/test-fails.cy.js
+        cy.on("fail", (e, runnable) => {
+          console.error("error", e);
+          console.error("runnable", runnable);
+
+          if (
+            e.name === "AssertionError" &&
+            e.message.includes("of 3 steps completed")
+          ) {
+            return false;
+          }
+          return true;
+        });
+
```

Comment:
This is a neat solution :+1: 

I was wondering if we could have also disabled this "steps" checks for dev with the `onlyOn` approach. Wdyt?

Reply:
@kelvah thank you, I am testing your suggestion https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/230, again there, I will run 10 attempts.

Reply:
Is possible to return true in the `.on("fail",...)` code part when the test is running on Mocked Env? I mean add this condition: `if (isEnvironmentType(EnvType.Dev))`

Reply:
Not sure @sjamboro I understand what do you mean. Do you mean the test should always fail, if this 'special handling of assertion error' happens on mocked environment?

Maybe you can share whole `cy.on("fail" ...` block you have in mind?

Reply:
I am working on that I have tried to simulate the error. I have rewritten the expectation like `4 of 3 steps completed` and I do not get the result as I expected.

Reply:
@jomarko do you try to simulate the error locally? I deleted the condition and it seems that the test is allways marked as successfull (no metter if I return `true` or `false`). See:

```
       cy.on("fail", (e, runnable) => {
          return true;
        }); 

        cy.ouiaId("steps-count", "QE/StackItem", { timeout: 10000 }).should(
          "have.text",
          "4 of 3 steps completed"
        );
```

Reply:
@jomarko I have included the code which has the expected behaviour (IIUC).

**In case I assert:** `cy.ouiaId("steps-count", "QE/StackItem", { timeout: 10000 }).should("have.text", "4 of 3 completed");`
A) **Mask error on DEV**
- the console contains the error log but the test is marked as succeeded.
- I am not sure if `console.error(...);` works here even if I delete them the console contains the error log and there is no double log.

B) **failed on MOCKED**
- the AssertionError alert is present and the test is marked as failed.

C) **the condition masked some problem**
- the whole message is `to have text 4 of 3 completed, but the text was 0 of 3 steps completed`
- the important part is that we assert `.should('have.text')` so if the element has text which includes `of 3 steps completed` the whole test is marked as succeeded

```
        cy.on("fail", (e, runnable) => {
          console.error("error", e);
          console.error("runnable", runnable);
      
          if (
            isEnvironmentType(EnvType.Dev) &&
            e.name === "AssertionError" &&
            e.message.includes("of 3 steps completed")
          ) {
            return true; //does not matter if true or false
          }
          throw e;
        });
```

Reply:
@sjamboro thank you for an investigation. I pushed your proposal in a separate commit.